.github/workflows/release.yml

```diff
@@ -0,0 +1,95 @@
+name: 'release'
+
+on:
+  push:
+    branches:
+    - 'main'
+
+jobs:
+  # build compiles the code and creates a pull request of the compiled result if
+  # there is a diff.
+  build:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: 'actions/checkout@v2'
+
+    - uses: 'actions/setup-node@v2'
+      with:
+        node-version: '12.x'
+
+    - name: 'npm build'
+      run: 'npm ci && npm run build'
+
+    - name: 'Commit'
+      run: |-
+        git config user.name "github-actions"
+        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git add dist/
+        git commit -m "chore: build dist"
+
+    - name: 'Create pull request'
+      uses: 'peter-evans/create-pull-request@v3'
```

Comment:
This is a 3rd party action so I am not sure if using the default GITHUB_TOKEN is a good idea since a bad release could do malicious activities on our repo? 

We could pin to a SHA or just remove this dependency and replace with a GH script?

Reply:
Making this a GitHub script is rather complicated. This doesn't just "create" a PR, it updates existing PRs too.

This isn't a new action (the old one used it), but you're right that the threat model is slightly different since we're using `GITHUB_TOKEN`.

I think pinning to a SHA is the best approach.

Reply:
Pinned to SHA.